### PR TITLE
GH-771: Fix N4JS_IDE launch configuration

### PR DESCRIPTION
**Pull request by lbeurerkellner**
Fixes https://github.com/eclipse/n4js/issues/771.

A previous misconfiguration of the included features lead to a validation issue of a dependency with regard to slf4j. This PR fixes this issue.

### CODE REVIEW

File: builds/org.eclipse.n4js.product.build/N4JS__IDE.launch [hunk 1]
```diff
@@ -66,6 +59,8 @@
 <setEntry value="org.eclipse.wst.xml_core.feature:default"/>
 <setEntry value="org.eclipse.wst.xml_ui.feature:default"/>
 <setEntry value="org.eclipse.wst.xml_userdoc.feature:default"/>
+<setEntry value="org.eclipse.epp.mpc:default"/>
```
Comment (qtran-n4) [inline builds/org.eclipse.n4js.product.build/N4JS__IDE.launch]: Why did you remove `org.eclipse.epp.mpc` bundle from the run configuration? This bundle allows market place. So theoretically, we can try the market place in the IDE started from within the development IDE. Practically, due to some other issues, this didn't work for me but still.

File: builds/org.eclipse.n4js.product.build/N4JS__IDE.launch [hunk 2]
```diff
@@ -66,6 +59,8 @@
...
+<setEntry value="org.eclipse.userstorage:default"/>
```
Comment (qtran-n4) [inline builds/org.eclipse.n4js.product.build/N4JS__IDE.launch]: This was required by `org.eclipse.epp.mpc` as far as I remember correctly.

**Comment by qtran-n4** [comment]
Can we list the required features (i.e. sdk) in the launch configuration instead of plugins ?

**Comment by qtran-n4** [comment]
We may consider to always start an N4JS IDE using the production definition (as a single source of truth) as discussed. See for instance https://www.modumind.com/2012/11/08/rcp-best-practices-always-run-code-using-a-product-configuration/

File: builds/org.eclipse.n4js.product.build/N4JS__IDE.launch (hunk 1 above)
Comment (lbeurerkellner) [reply builds/org.eclipse.n4js.product.build/N4JS__IDE.launch]: According to my tests, this configuration allows for the use of the marketplace in a smith-launched IDE.

Comment (qtran-n4) [reply builds/org.eclipse.n4js.product.build/N4JS__IDE.launch]: Exactly, that's why it may be a good idea to keep it there?

Comment (qtran-n4) [reply builds/org.eclipse.n4js.product.build/N4JS__IDE.launch]: Ah it is kept. The diff view is a bit confusing.
